param/_utils.py

```diff
@@ -219,6 +227,34 @@ def iscoroutinefunction(function):
         inspect.iscoroutinefunction(function)
     )
 
+async def _to_thread(func, /, *args, **kwargs):
+    """
+    Polyfill for asyncio.to_thread in Python < 3.9
+    """
+    loop = asyncio.get_running_loop()
+    ctx = contextvars.copy_context()
+    func_call = functools.partial(ctx.run, func, *args, **kwargs)
+    return await loop.run_in_executor(None, func_call)
+
+async def _to_async_gen(sync_gen):
+    done = object()
+
+    def safe_next():
+        # Converts StopIteration to a sentinal value to avoid:
+        # TypeError: StopIteration interacts badly with generators and cannot be raised into a Future
+        try:
+            return next(sync_gen)
+        except StopIteration:
+            return done
+
+    while True:
+        if sys.version_info >= (3, 9):
```

Comment:
If this is gonna be in Param 2.1, maybe we should drop Python 3.8 support?

Reply:
I would like that.

Reply:
Will leave this in and then make a final decision on removing this or not before release.